torch_geometric/utils/scatter.py

```diff
@@ -1,64 +1,79 @@
+
+    # Broadcast `index`:
+    dim = src.dim() + dim if dim < 0 else dim
+    if index.dim() == 1:
+        for _ in range(0, dim):
+            index = index.unsqueeze(0)
+    for _ in range(index.dim(), src.dim()):
+        index = index.unsqueeze(-1)
+    index = index.expand(src.size())
+
+    include_self = out is not None
+
+    if out is None:  # Generate `out` if not given:
+        size = list(src.size())
+        if dim_size is not None:
+            size[dim] = dim_size
+        elif index.numel() > 0:
+            size[dim] = int(index.max()) + 1
+        else:
+            size[dim] = 0
+        out = src.new_zeros(size)
 
-experimental = os.environ.get('PYG_EXPERIMENTAL', False)
+    out = out.scatter_reduce_(dim, index, src, reduce,
+                              include_self=include_self)
 
-# Requires PyTorch >= 1.12
-if experimental and get_pytorch112:  # pragma: no cover
-    warnings.filterwarnings('ignore', '.*scatter_reduce.*')
+    # TODO: For now, we need the clone here since otherwise, autograd will
+    # complain about inplace modifications.
+    # Reference: https://github.com/pyg-team/pytorch_geometric/pull/5120
+    out = out.clone()
 
-    def scatter(src: Tensor, index: Tensor, dim: int = -1,
+    return out
+
+
+def _scatter_ts(src: Tensor, index: Tensor, dim: int = -1,
                 out: Optional[Tensor] = None, dim_size: Optional[int] = None,
                 reduce: str = 'sum') -> Tensor:
+    return torch_scatter.scatter(src, index, dim, out, dim_size, reduce)
 
-        reduce = 'sum' if reduce == 'add' else reduce
-        reduce = 'prod' if reduce == 'mul' else reduce
-        reduce = 'amin' if reduce == 'min' else reduce
-        reduce = 'amax' if reduce == 'max' else reduce
-
-        # Broadcast `index`:
-        dim = src.dim() + dim if dim < 0 else dim
-        if index.dim() == 1:
-            for _ in range(0, dim):
-                index = index.unsqueeze(0)
-        for _ in range(index.dim(), src.dim()):
-            index = index.unsqueeze(-1)
-        index = index.expand(src.size())
-
-        include_self = out is not None
-
-        if out is None:  # Generate `out` if not given:
-            size = list(src.size())
-            if dim_size is not None:
-                size[dim] = dim_size
-            elif index.numel() > 0:
-                size[dim] = int(index.max()) + 1
-            else:
-                size[dim] = 0
-            out = src.new_zeros(size)
-
-        out = out.scatter_reduce_(dim, index, src, reduce,
-                                  include_self=include_self)
-
-        # TODO For now, we need the clone here since otherwise, autograd will
-        # complain about inplace modifications.
-        # Reference: https://github.com/pyg-team/pytorch_geometric/pull/5120
-        out = out.clone()
-
-        return out
-
-else:
-    import torch_scatter
-
-    def scatter(src: Tensor, index: Tensor, dim: int = -1,
+
+class ScatterImpl(torch.nn.Module):
```

Comment:
Sorry stupid question - why do we want this to be a module?

Reply:
The short answer is that we need this for `torch.jit.script` to work with `scatter` function. I just followed JIT troubleshooting scheme.

Reply:
Got it, thanks for the details